TasksApp/TasksApp/WeeklyTaskService.cs

```diff
@@ -0,0 +1,162 @@
+    {
+        private readonly string _path = @"C:\Users\MONCEY\source\repos\HomeworkRepository\TasksApp\Tasks.t";
+        private readonly List<WeeklyTask> _listOfTasks = new();
+
+        internal void ExportTasks()
+        {
+            using var writer = new BinaryWriter(File.Open(_path, FileMode.OpenOrCreate));
+
+            foreach (var t in _listOfTasks)
+            {
+                writer.Write(t.ToSave());
+            }
+        }
+
+        internal void ImportTasks()
+        {
+            using var reader = new BinaryReader(File.Open(_path, FileMode.OpenOrCreate));
+            while (reader.PeekChar() > -1)
+            {
+                AddTask(reader.ReadString());
+            }
+        }
+
+        internal void AddTask(string task)
+        {
+            var tempTasks = GetTaskFromString(task);
+            tempTasks.Id = _listOfTasks.Count;
+
+            _listOfTasks.Add(tempTasks);
+        }
+
+        internal void PrintAllTasks()
+        {
+            PrintIsEmpty();
+            PrintWithForeach(_listOfTasks);
+        }
+
+        internal void HandlerFilter(string filter)
+        {
+            string[] parts = filter.Split(' ');
+
+            if (parts.Length == 3 && parts[0] == "filter")
+            {
+                if (parts[1] == "priority")
+                {
+                    PrintFilterPriority(GetTasksPriority(parts[2]));
+                }
+                else if (parts[1] == "date")
+                {
+                    PrintFilterDate(GetDateTime(parts[2]));
+                }
+                else
+                {
+                    throw new Exception("Wrong filter format !");
+                }
+            }
+            else
+            {
+                throw new Exception("Wrong filter format !");
+            }
+        }
+
+        internal void EditTask(int id, string newtask) => _listOfTasks[id] = GetTaskFromString(newtask);
+
+        internal void DeleteTask(int id)
+        {
+            _listOfTasks.RemoveAt(id);
+            ReassignmentId();
+        }
+
+        internal string AlarmTask(int id) => _listOfTasks[id].GetAlarm();
+
+        internal bool CorrectId(int id) => id < _listOfTasks.Count;
+
+        private void PrintIsEmpty()
+        {
+            Console.Clear();
+
+            if (_listOfTasks.Count == 0)
+            {
+                Console.WriteLine("There are no tasks!");
+            }
+        }
+
+        private void FilterIsEmpty(IEnumerable<WeeklyTask> list)
+        {
+            Console.Clear();
+
+            if (!list.Any())
+            {
+                Console.WriteLine("There are no tasks!");
+            }
+        }
+
+        private void PrintFilterDate(DateTime date)
+        {
+            var list = _listOfTasks.Where(d => (d as PriorityTask)?.Date > date).Concat(_listOfTasks.Where(d => (d as RegularTask)?.Date > date));
+
+            FilterIsEmpty(list);
+            PrintWithForeach(list);
```

Comment:
`FilterIsEmpty(list);` should be called inside `PrintWithForeach` I think - they always go together, right?